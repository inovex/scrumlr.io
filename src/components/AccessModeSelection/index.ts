import AccessModeSelection from './AccessModeSelection';
export default AccessModeSelection;
export * from './AccessModeSelection';
