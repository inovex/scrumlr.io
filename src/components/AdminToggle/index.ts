export * from './AdminToggle';

import AdminToggle from './AdminToggle';
export default AdminToggle;
