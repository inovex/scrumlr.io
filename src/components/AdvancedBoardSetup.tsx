import React from 'react';
import Typography from '@material-ui/core/Typography';
import FormGroup from '@material-ui/core/FormGroup';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import { useTranslation } from 'react-i18next';

export interface AdvancedBoardSetupProps {
  admissionControl: boolean;
  encryptedData: boolean;
  onSetAdmissionControl: (enabled: boolean) => void;
  onSetEncryptedData: (enabled: boolean) => void;
}

export const AdvancedBoardSetup: React.FC<AdvancedBoardSetupProps> = ({ admissionControl, encryptedData, onSetAdmissionControl, onSetEncryptedData }) => {
  const { t } = useTranslation('advancedBoardSetup');

  return (
    <>
      <ExpansionPanel>
        <ExpansionPanelSummary>
          <Typography>{t('heading')}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div>
            <Typography>{t('helper')}</Typography>
            <FormGroup>
              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={
                  <Switch
                    checked={admissionControl}
                    onChange={(event) => {
                      onSetAdmissionControl(event.target.checked);
                    }}
                  />
                }
                label={t('admissionControl.heading')}
              />
              <Typography color="textSecondary">{t('admissionControl.helper')}</Typography>

              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={<Switch checked={encryptedData} onChange={(event) => onSetEncryptedData(event.target.checked)} disabled={!admissionControl} />}
                label={t('encryptData.heading')}
              />
              <Typography color="textSecondary">{t('encryptData.helper')}</Typography>
            </FormGroup>
          </div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </>
  );
};

export default AdvancedBoardSetup;
