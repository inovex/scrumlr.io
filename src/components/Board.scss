@import "common";

.board {
  --n: 1;

  display:grid;
  grid-auto-flow: column;
  grid-auto-columns: calc(100%/min(var(--real-columns), var(--n)));
  overflow-x:auto;
  scroll-snap-type: x mandatory;

  height: 100vh;
}

@for $i from 1 through 12 {
  @media screen and (min-width: $i*$min-column-width) {
    .board {
      --n: #{$i};
    }
  }
}