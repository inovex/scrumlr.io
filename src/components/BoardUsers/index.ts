export * from "./BoardUsers";
export * from "./UserAvatar";
