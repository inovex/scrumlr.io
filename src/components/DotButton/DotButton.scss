@import "constants/style";

.dot-button {
  border-radius: 100%;
  border: none;
  padding: 0;
  font-size: $text-size--medium;
  line-height: 28px;
  font-weight: 600;
  cursor: pointer;
  outline: none;
}
.dot-button:disabled {
  cursor: default;
}
.dot-button:not([disabled]):active {
  transform: scale(0.9);
}
