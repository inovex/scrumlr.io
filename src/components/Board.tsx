import React from 'react';
import './Board.scss';
import {getColorClassName} from "../constants/colors";

const Board = ({ children }: any) => {
    const colors = React.Children.map(children, (child) => child.props.color);

    return (
        <>
            <style>
                {`.board { --board__columns: ${React.Children.count(children)} }`}
            </style>
            <div className="board">
                <div className={`board__spacer-left ${getColorClassName(colors[0])}`} />
                    {children}
                <div className={`board__spacer-right ${getColorClassName(colors[colors.length - 1])}`} />
            </div>
        </>
    )
};

export default Board;