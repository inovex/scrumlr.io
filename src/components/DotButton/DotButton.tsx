import {FC, PropsWithChildren} from "react";
import classNames from "classnames";
import "./DotButton.scss";

type DotButtonProps = {
  className?: string;
  disabled?: boolean;
  onClick?: () => void;
  title?: string;
  onAnimationEnd?: () => void;
};

export const DotButton: FC<PropsWithChildren<DotButtonProps>> = (props) => (
  <button
    className={classNames("dot-button", props.className)}
    disabled={props.disabled}
    onClick={(e) => {
      e.stopPropagation();
      props.onClick?.();
    }}
    title={props.title}
    onAnimationEnd={props.onAnimationEnd}
  >
    {props.children}
  </button>
);
