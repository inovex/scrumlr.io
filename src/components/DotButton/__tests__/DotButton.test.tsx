import {fireEvent} from "@testing-library/react";
import {DotButton} from "components/DotButton";
import {render} from "testUtils";

describe("DotButton", () => {
  test("should render correctly", () => {
    const dotButton = render(
      <DotButton>
        <span>test-text</span>
      </DotButton>
    );
    expect(dotButton.container).toMatchSnapshot();
  });

  test("should render correctly with additional classname", () => {
    const dotButton = render(
      <DotButton className="test-classname">
        <span>test-text</span>
      </DotButton>
    );
    expect(dotButton.container).toMatchSnapshot();
  });

  test("should call 'onClick' if given", () => {
    const mockOnClick = jest.fn();
    const dotButton = render(<DotButton onClick={mockOnClick} />);
    fireEvent.click(dotButton.container.getElementsByTagName("button")[0]);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
