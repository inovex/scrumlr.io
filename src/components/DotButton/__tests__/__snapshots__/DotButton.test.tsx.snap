// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotButton should render correctly 1`] = `
<div>
  <button
    class="dot-button"
  >
    <span>
      test-text
    </span>
  </button>
</div>
`;

exports[`DotButton should render correctly with additional classname 1`] = `
<div>
  <button
    class="dot-button test-classname"
  >
    <span>
      test-text
    </span>
  </button>
</div>
`;
